folio-token/src/main/java/org/folio/spring/context/ExecutionContextBuilder.java

```diff
@@ -0,0 +1,144 @@
+package org.folio.spring.context;
+
+import static java.util.Collections.singleton;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.folio.spring.kafka.KafkaUtils.getHeaderValue;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import lombok.With;
+import org.folio.spring.FolioExecutionContext;
+import org.folio.spring.FolioModuleMetadata;
+import org.folio.spring.integration.XOkapiHeaders;
+import org.folio.spring.model.SystemUser;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class ExecutionContextBuilder {
+
+  private final FolioModuleMetadata moduleMetadata;
+
+  public Builder builder() {
+    return new Builder(moduleMetadata);
+  }
+
+  public FolioExecutionContext forMessageHeaders(MessageHeaders headers) {
```

Comment:
I think this is out of scope for the module

Reply:
removed